tests/unit/test_repos_repo.py

```diff
@@ -526,6 +526,35 @@ def test_create_tag_that_is_not_lightweight(self):
             },
         )
 
+    def test_create_tag_that_is_not_lightweight_and_forced(self):
+        """Verify we can create an annotated tag."""
+        self.instance.create_tag(
+            tag='tag-name',
+            message='message',
+            sha='my-sha',
+            obj_type='commit',
+            tagger={'name': 'Ian Cordasco',
+                    'email': 'example@example.com',
+                    'date': '2015-11-01T12:16:00Z'},
+            lightweight=False,
+            update=True
+        )
+
+        self.post_called_with(
+            url_for('git/tags'),
+            data={
+                'tag': 'tag-name',
+                'message': 'message',
+                'object': 'my-sha',
+                'type': 'commit',
+                'tagger': {
+                    'name': 'Ian Cordasco',
+                    'email': 'example@example.com',
+                    'date': '2015-11-01T12:16:00Z',
+                },
+            },
```

Comment:
Unless I'm missing something, this doesn't validate that the call went through `update_ref` instead of `create_ref`. Is there a way you could validate that instead?

Reply:
Why does the internal call chain matter?  This is a behavioral test that verifies that the given input produces the desired result.

Reply:
Because as a unit test, we want to ensure the right decision path is used, and that the code path you've added is exercised. We should at least see `patch` called instead of `post`.